setup.cfg

```diff
@@ -194,3 +194,36 @@ source_modules =
 forbidden_modules =
     openedx_learning.apps
 allow_indirect_imports = True
+
+[importlinter:contract:4]
+name = Low-level apps should not depend on high-level apps
+type = layers
+layers =
+    # Layers from high-level to low-level. Imports should only occur from higher to lower.
+    cms.lib.xblock.upstream_sync | openedx.core.djangoapps.content.search | openedx.core.djangoapps.olx_rest_api
+    openedx.core.djangoapps.content_libraries
+    openedx.core.djangoapps.content_staging
+    openedx.core.djangoapps.xblock
+    openedx.core.lib.xblock_serializer
+    openedx.core.djangoapps.content_tagging
+ignore_imports =
+    # Test code can break these layering rules
+    **.tests.** -> **
+
+    # FIXME: the exceptions below are from before we added this import linting rule. Should refactor to eliminate them.
```

Comment:
contentstore in general is a dumping ground for everything Studio-related. Would love to embark on a refactoring of it one day.